apps/gitness/src/pages-v2/pull-request/pull-request-conversation.tsx

```diff
@@ -346,137 +492,64 @@ export default function PullRequestConversationPage() {
     pullReqMetadata?.merged
   ])
 
-  const handleAddReviewer = (id?: number) => {
-    reviewerAddPullReq({ repo_ref: repoRef, pullreq_number: prId, body: { reviewer_id: id } })
-      .then(() => refetchReviewers())
-      .catch(error => setAddReviewerError(error.message))
-  }
+  const handleAddReviewer = useCallback(
+    (id?: number) => {
+      reviewerAddPullReq({
+        repo_ref: repoRef,
+        pullreq_number: prId,
+        body: { reviewer_id: id }
+      })
+        .then(() => refetchReviewers())
+        .catch(error => setAddReviewerError(error.message))
+    },
+    [repoRef, prId, refetchReviewers]
+  )
+
   const handleDeleteReviewer = (id: number) => {
     reviewerDeletePullReq({ repo_ref: repoRef, pullreq_number: prId, pullreq_reviewer_id: id })
       .then(() => refetchReviewers())
       .catch(error => setRemoveReviewerError(error.message))
   }
 
-  const onPRStateChanged = useCallback(() => {
+  const handleRefetchData = useCallback(() => {
     refetchCodeOwners()
     refetchPullReq()
     refetchActivities()
-  }, [refetchCodeOwners, repoRef, handleAddReviewer, handleDeleteReviewer]) // eslint-disable-line react-hooks/exhaustive-deps
-
-  const handleMerge = (method: string) => {
-    const payload: OpenapiMergePullReq = {
-      method: method as EnumMergeMethod,
-      source_sha: pullReqMetadata?.source_sha,
-      bypass_rules: checkboxBypass,
-      dry_run: false
-      // message: data.commitMessage
-    }
-    mergePullReqOp({ body: payload, repo_ref: repoRef, pullreq_number: prId }).then(() => {
-      onPRStateChanged()
-      setRuleViolationArr(undefined)
-    })
-    //todo: add catch t o show errors
-    // .catch(exception => showError(getErrorMessage(exception)))
-  }
-
-  const handlePrState = (state: string) => {
-    const payload: OpenapiStatePullReqRequest = {
-      ...(state === 'draft' && { is_draft: true }),
-      state: state === 'draft' ? 'open' : (state as EnumPullReqState)
-    }
-    statePullReq({ body: payload, repo_ref: repoRef, pullreq_number: prId }).then(() => {
-      onPRStateChanged()
-      setRuleViolationArr(undefined)
-    })
-  }
-
-  const handleRebaseBranch = () => {
-    const payload: RebaseBranchRequestBody = {
-      base_branch: pullReqMetadata?.target_branch,
-      bypass_rules: true,
-      dry_run_rules: false,
-      head_branch: pullReqMetadata?.source_branch,
-      head_commit_sha: pullReqMetadata?.source_sha
-    }
-
-    rebaseBranch({ body: payload, repo_ref: repoRef }).then(
-      () => {
-        onPRStateChanged()
+  }, [refetchCodeOwners, refetchPullReq, refetchActivities])
+
+  const handleMerge = useCallback(
+    (method: string) => {
+      const payload: OpenapiMergePullReq = {
+        method: method as EnumMergeMethod,
```

Comment:
You can get rid of `as EnumMergeMethod` by just setting the arg as that type (with a corresponding change in `getMockPullRequestActions`.

Also, `EnumMergeMethod` really annoys me since it is not an enum, but just a type.
```suggestion
    (method: EnumMergeMethod) => {
      const payload: OpenapiMergePullReq = {
        method,
```